octobot/community/authentication.py

```diff
@@ -0,0 +1,177 @@
+#  Drakkar-Software OctoBot
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import functools
+import time
+import requests
+
+import octobot.constants as constants
+import octobot_commons.constants as commons_constants
+import octobot_commons.config_manager as config_manager
+import octobot_commons.logging as bot_logging
+
+
+def authenticated(func):
+    @functools.wraps(func)
+    def wrapped(self, *args, **kwargs):
+        self.ensure_token_validity()
+        return func(self, *args, **kwargs)
+    return wrapped
+
+
+class CommunityAuthentication:
+    """
+    Authentication utility
+    """
+    ALLOWED_TIME_DELAY = 1 * commons_constants.MINUTE_TO_SECONDS
+    AUTHORIZATION_HEADER = "Authorization"
+
+    def __init__(self, authentication_url, username=None, password=None, config=None):
+        self.logger = bot_logging.get_logger(self.__class__.__name__)
+        self.authentication_url = authentication_url
+        self.refresh_token = None
+        self.edited_config = config
+
+        self._auth_token = None
+        self._expire_at = None
+        self._session = requests.Session()
+        self._cache = {}
+
+        if None not in (username, password):
```

Comment:
I'm not a big fan of `None not in (username, password)`. I would prefer `username and password`

Reply:
The problem with this check is that it prevents the opportunity for an empty password / username auth

Reply:
Do you think we should able to use empty login username and password ?

Reply:
I just don't see why we should forbid